pkg/commands/entrypoints/entrypoints.go

```diff
@@ -233,41 +297,92 @@ func (c *EntrypointsCommand) Process(ctx context.Context) error {
 		entrypointDirs = files
 	}
 
-	logger.DebugContext(ctx, "target directories", "target_directories", entrypointDirs)
+	logger.DebugContext(ctx, "calculated entrypoints and removed dirs",
+		"entrypoints", entrypointDirs,
+		"removed_dirs", removedDirs,
+	)
 
-	if err := c.writeOutput(entrypointDirs); err != nil {
-		return fmt.Errorf("failed to write output: %w", err)
+	return entrypointDirs, removedDirs, nil
+}
+
+// findRemovedDirs tests if any directories were removed from the filesystem and returns the list of removed dirs.
+func (c *EntrypointsCommand) findRemovedDirs(dirs []string) ([]string, error) {
+	removed := make([]string, 0)
+
+	for _, dir := range dirs {
+		_, err := os.Stat(dir)
+		// there was an error testing if dir exists
+		if err != nil && !errors.Is(err, fs.ErrNotExist) {
+			return nil, fmt.Errorf("failed to check if dir exists: %w", err)
+		}
+
+		if errors.Is(err, fs.ErrNotExist) {
+			removed = append(removed, dir)
+		}
 	}
 
-	return nil
+	return removed, nil
 }
 
-// writeOutput writes the command output.
-func (c *EntrypointsCommand) writeOutput(dirs []string) error {
-	cwd, err := c.WorkingDir()
-	if err != nil {
-		return fmt.Errorf("failed to get current working directory: %w", err)
+// processDestroyMetaValues processes the user supplied meta values for directories to destroy.
+func (c *EntrypointsCommand) processDestroyMetaValues(cwd string, dirs []string, metaValues modifiers.MetaValues) ([]string, error) {
+	destroyDirs := make([]string, 0)
+
+	metaDestroyDirs, ok := metaValues["GUARDIAN_DESTROY"]
```

Comment:
Maybe make `"GUARDIAN_DESTROY"` a top-level const, since it will probably be referenced elsewhere